tests/stress_tests/memcheck_tests/tests_pipelines/tests_pipelines.cpp

```diff
@@ -11,130 +11,201 @@
-        test_cur_vmpeak -= vmsize_before_test;                                              \
-        test_cur_vmrss -= vmrss_before_test;                                                \
-        test_cur_vmhwm -= vmrss_before_test;
-
-#define log_debug_ref_record_for_test(test_name)                                                            \
-        log_debug("Record to update reference config: "                                                           \
-                  << "<model path=\"" + model_name + "\"" + " test=\"" + test_name + "\" device=\"" +       \
-                  target_device +                                                                           \
-                  "\" vmsize=\"" + std::to_string((int) (test_cur_vmsize * REPORTING_THRESHOLD)) +          \
-                  "\" vmpeak=\"" + std::to_string((int) (test_cur_vmpeak * REPORTING_THRESHOLD)) +          \
-                  "\" vmrss=\"" + std::to_string((int) (test_cur_vmrss * REPORTING_THRESHOLD)) +            \
-                  "\" vmhwm=\"" + std::to_string((int) (test_cur_vmhwm * REPORTING_THRESHOLD)) + "\" />");
-
-#define log_info_ref_mem_usage()                                                                \
-        log_info("Reference values of virtual memory consumption:");                            \
-        log_info("VMRSS\t\tVMHWM\t\tVMSIZE\t\tVMPEAK");                                               \
-        log_info(ref_vmrss << "\t\t" << ref_vmhwm << "\t\t" << ref_vmsize << "\t\t" << ref_vmpeak);
-
-#define log_info_cur_mem_usage()                                                                                    \
-        log_info("Current values of virtual memory consumption:");                                                  \
-        log_info("VMRSS\t\tVMHWM\t\tVMSIZE\t\tVMPEAK");                                                                   \
-        log_info(test_cur_vmrss << "\t\t" << test_cur_vmhwm << "\t\t" << test_cur_vmsize << "\t\t" << test_cur_vmpeak);
+
+/**
+ * @brief Class response for encapsulating measure and measurements printing
+ */
+class MemCheckPipeline {
+private:
+    std::array<long, MeasureValueMax> measures;            // current measures
+    std::array<long, MeasureValueMax> start_measures;      // measures before run (will be used as baseline)
+public:
+    MemCheckPipeline() {
+        start_measures[VMRSS] = (long) getVmRSSInKB();
+        start_measures[VMHWM] = start_measures[VMRSS];
+        start_measures[VMSIZE] = (long) getVmSizeInKB();
+        start_measures[VMPEAK] = start_measures[VMSIZE];
+        start_measures[THREADS] = (long) getThreadsNum();
+
+        std::copy(start_measures.begin(), start_measures.end(), measures.begin());
+    }
+
+    ~MemCheckPipeline() {
+        // message required for DB data upload
+        log_info("Current values of virtual memory consumption:");
+        print_measures();
```

Comment:
The idea is to print the latest measurement without any another measure. The first measure will be done in ctor (using `start_measures`), so it will print zeros (aligned on baseline). In this case I don't see any dangerous cases

Reply:
Perhaps it isn't intuitive that in scope of measures there shouldn't be presented any deleting of objects defined before `MemCheckPipeline` creation because it may causes negative values while print. I propose it should be documented

Reply:
I documented `MemCheckPipeline` to prevent any misunderstanding while use it